### DISCUSSION TITLE
Ability to control changing transforms or map keyboard buttons and arrows

### DISCUSSION THREAD

**Discussion by cpyarger** [cpyarger/obs-midi#105] [Ideas]
> One handy feature would be able to use either several midi buttons or a fader/knob to control transform X/Y, zoom, etc so you can control position and size of sources in a scene from the midi controller. A fader would be a single axis.
> 
> Another option would be to use a fader or buttons to replicate arrow keys for toggling between items or editing transforms. A user would still have to click on what they want before using the buttons.
> 
> Keep up the good work!
From  @countdookuw in Alzy#28

**Comment by traitzco** [comment]
Yes! I would love this. Being able to control any parameter - even controlling rotation that some plugins offer

**Comment by traitzco** [comment]
OSC features would be great too but that's a separate issue (or separate project) all-together

**Reply by cpyarger** [reply]
Yes, I agree, I have an RFC and PR in for obs-studio on a controls mapping system which would greatly simplify building plugins for controlling obs or having obs control other things, If that ever gains any traction, I am planning on writing a bunch of plugins for various control protocols, OSC, Artnet, MIDI, and MQTT to name a few.

**Comment by cpyarger** [comment]
Adding source Rotation to a release building now.
